packages/orval/src/utils/package-json.ts

```diff
@@ -38,7 +38,7 @@ const maybeReplaceCatalog = async (
       ...Object.entries(pkg.dependencies ?? {}),
       ...Object.entries(pkg.devDependencies ?? {}),
       ...Object.entries(pkg.peerDependencies ?? {}),
-    ].some(([key]) => key.startsWith('catalog:'))
+    ].some(([, value]) => typeof value === 'string' && value.startsWith('catalog:'))
```

Comment:
You can use `isString` instead 👍 

```suggestion
    ].some(([, value]) => isString(value) && value.startsWith('catalog:'))
```

This can be imported from core.

```diff
+ import { isString } from '@orval/core';
```

Reply:
updated